android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -656,7 +664,27 @@ public void applySceneUpdates() {
      * @param use Whether to use a cached OpenGL state; false by default
      */
     public void useCachedGlState(boolean use) {
-        nativeUseCachedGlState(use);
+        nativeUseCachedGlState(mapPointer, use);
+    }
+
+
+    // Package private methods
+    // =======================
+
+    void removeDataSource(long sourcePtr) {
+        nativeRemoveDataSource(mapPointer, sourcePtr);
+    }
+
+    void clearDataSource(long sourcePtr) {
+        nativeClearDataSource(mapPointer, sourcePtr);
+    }
+
+    void addFeature(long sourcePtr, double[] coordinates, int[] rings, String[] properties) {
+        nativeAddFeature(mapPointer, sourcePtr, coordinates, rings, properties);
+    }
+
+    void addGeoJson(long sourcePtr, String geoJson) {
+        nativeAddGeoJson(mapPointer, sourcePtr, geoJson);
```

Comment:
this should not require `mapPointer`. Also the corresponding jni call does not match in the `jniExports.cpp`.